### PR TITLE
use modern API in userauth_keyboard_interactive()

### PR DESCRIPTION
**Pull request by xalopp**
This refactors SSH_MSG_USERAUTH_INFO_REQUEST processing in `userauth_keyboard_interactive()` in order to improve robustness, correctness and readability or the code.

In order to be able to do unit testing, I moved some portion of the code from `src/userauth.c` to  `src/userauth_kbd_packet.c`

### CODE REVIEW

File: tests/test_keyboard_interactive_auth_info_request.c [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+        {
+            "<"
+            "\0\0\0\x01"
+            "\0\0\0\0"
+            "\0\0\0\0"
+            "\0\0\0\0", 17,
+            {FAIL, -38,
+            "Unable to decode keyboard-interactive number of "
+            "keyboard prompts"}},
+        /* to many prompts */
```
Comment (willco007) [inline tests/test_keyboard_interactive_auth_info_request.c:118]: typo, "to" should be "too"

**Comment by willco007** [comment]
Looks like there is a linker error that needs to be fixed up; but at first glance this looks great!

**Comment by willco007** [comment]
The makefile needs to be updated to include `userauth_kbd_packet.c`, should fix it up.

**Comment by xalopp** [comment]
> The makefile needs to be updated to include `userauth_kbd_packet.c`, should fix it up.

My bad. Totally forgot the autotools build.

**Comment by willco007** [comment]
@xalopp thank you for your continued patience working on the Windows builds. 👍

**Comment by xalopp** [comment]
@willco007 finally, the build is green :smile_cat: 

In order to be able to do unit testing, I move some code to  `src/userauth_kbd_packet.c`. This currently adds  `userauth_keyboard_interactive_decode_info_request` to the symbol list of `libssh2.a` / `libssh2.so`. Since it's not part of `libssh2.h`, it's a "unofficial" function that will be part of the API.  Not sure if this is a issue.

**Comment by willco007** [comment]
I've looked at this and it looks good to me; anyone else have time to take a look at this?

**Pull request merged**